crates/matrix-sdk-ui/src/room_list_service/room_list.rs

```diff
@@ -142,8 +144,41 @@ impl RoomList {
         let stream = stream! {
             loop {
                 let filter_fn = filter_fn_cell.take().await;
-                let (values, stream) = list
-                    .room_list_stream()
+                let (raw_values, mut raw_stream) = list.room_list_stream();
+                let mut raw_current_values = raw_values.clone();
+                let client = client.clone();
+                let raw_stream_with_recv = stream! {
+                    loop {
+                        let mut roominfo_update_recv = client.roominfo_update_recv();
```

Comment:
Instead of passing the whole `Client` to this function, we could pass it the receiver, and clone it (or not, if not necessary) here.

In general, it's slightly nicer to pass the minimal amount of data as parameters:

- it makes testing easier, as we don't need to mock extra data all over the place,
- it shows precisely what's being used by the called function ("callee")